### PR TITLE
Fix filtering for bookstore example

### PR DESCRIPTION
**Pull request by shadikhani**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/bookstore-example/45)

### CODE REVIEW

File: src/main/java/org/vaadin/example/bookstore/ui/login/LoginScreen.java [hunk 1]
```diff
@@ -11,10 +11,10 @@
 import com.vaadin.flow.router.PageTitle;
 import com.vaadin.flow.router.Route;
 import com.vaadin.flow.router.RouteConfiguration;
-import com.vaadin.samples.bookstore.authentication.AccessControl;
-import com.vaadin.samples.bookstore.authentication.AccessControlFactory;
-import com.vaadin.samples.bookstore.ui.AdminView;
-import com.vaadin.samples.bookstore.ui.MainLayout;
+import org.vaadin.example.bookstore.authentication.AccessControl;
+import org.vaadin.example.bookstore.authentication.AccessControlFactory;
+import org.vaadin.example.bookstore.ui.AdminView;
```
[the remark this answers is not in the document]
Comment (shadikhani) [reply src/main/java/org/vaadin/example/bookstore/ui/login/LoginScreen.java]: Done.

File: src/main/java/org/vaadin/example/bookstore/ui/login/LoginScreen.java [hunk 2]
```diff
@@ -11,10 +11,10 @@
...
+import org.vaadin.example.bookstore.ui.MainLayout;
```
[the remark this answers is not in the document]
Comment (shadikhani) [reply src/main/java/org/vaadin/example/bookstore/ui/login/LoginScreen.java]: Done.

File: src/main/java/org/vaadin/example/bookstore/ui/login/LoginScreen.java [hunk 3]
```diff
@@ -11,10 +11,10 @@
 import com.vaadin.flow.router.PageTitle;
 import com.vaadin.flow.router.Route;
 import com.vaadin.flow.router.RouteConfiguration;
```
[the remark this answers is not in the document]
Comment (shadikhani) [reply src/main/java/org/vaadin/example/bookstore/ui/login/LoginScreen.java]: Done.

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java [hunk 4]
```diff
@@ -72,15 +71,17 @@ public HorizontalLayout createTopBar() {
...
 
         newProduct = new Button("New product");
+        // Setting theme variant of new production button to LUMO_PRIMARY that
+        // changes its background color to blue and its text color to white
         newProduct.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
         newProduct.setIcon(VaadinIcon.PLUS_CIRCLE.create());
         newProduct.addClickListener(click -> viewLogic.newProduct());
-        // CTRL+N will create a new window which is unavoidable
+        // A shortcut to click the new product button by pressing ctrl + N
         newProduct.addClickShortcut(Key.KEY_N, KeyModifier.ALT);
```
Comment (samie) [inline src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java]: `KeyModifier.ALT` but the comment says `ctrl`

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java [hunk 5]
```diff
@@ -94,39 +95,73 @@ public void showError(String msg) {
...
+     * 
+     * @param product
+     */
     public void editProduct(Product product) {
         showForm(product != null);
         form.editProduct(product);
     }
 
+    /**
+     * Shows the new product form
```
Comment (samie) [inline src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java]: The logic seems to be "show or hide".

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java [hunk 6]
```diff
@@ -94,39 +95,73 @@ public void showError(String msg) {
...
+     * Updates a product
+     * 
+     * @param product
+     */
     public void updateProduct(Product product) {
         dataProvider.save(product);
     }
 
+    /**
+     * Removes a product
```
Comment (samie) [inline src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java]: I still feel this should explain "removed where". Like "removes the Product from 
(commented in #42) Same goes for update, edit, select.

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java [hunk 7]
```diff
@@ -94,39 +95,73 @@ public void showError(String msg) {
...
+     * Enables/Disables the new product button.
+     *
+     * @param enabled
+     */
     public void setNewProductEnabled(boolean enabled) {
         newProduct.setEnabled(enabled);
     }
 
+    /**
+     * Clears the fields in the product form.
```
Comment (samie) [inline src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java]: Are you sure? This looks like it unselects the item from Grid.

File: src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java [hunk 8]
```diff
@@ -94,39 +95,73 @@ public void showError(String msg) {
         Notification.show(msg);
     }
 
-    public void showSaveNotification(String msg) {
+    /**
+     * shows a message in a notification.
```
Comment (samie) [inline src/main/java/org/vaadin/example/bookstore/ui/inventory/InventoryView.java]: See comments in #42

**Pull request closed without merging**